notebooks/stitches_dev/Notebook6_throwout_Duplicates-across-ensemble-members.Rmd

```diff
@@ -0,0 +1,297 @@
+
+Note that because we want to enforce this across targets, it's really not parallellizable except across _experiments_. Also in general, with the way the code has been written, it's probably best to call it separately for each experiment of interest being worked with. Just because this whole duplication elimination stuff is not optional and it wouldn't make sense that points used in the construction of a new ssp245 realization couldn't be used in the construction of a new ssp585. 
+
+# Set Up 
+```{r, message=FALSE, warning=FALSE}
+# required packages
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+library(knitr)
+
+
+# The load the functions we will want to use, these are currently written in R and will be translated into python. 
+source("functions/nearest_neighbor_matching.R") # the function match_neighborhood is defined here
+source("functions/stitching_functions.R")  # the function stitch_global_mean is definend here 
+source("functions/permuting_matches.R")
+source("functions/gridded_recipe.R")
+```
+
+
+Load the inputs. 
+
+```{r}
+# Time series of the raw global mean temperature anomaly, this is the data that is 
+# going to be stitched together. Historical data is pasted into every scenario. No
+# smoothing has been done.
+tgav_data <- read.csv("inputs/main_raw_pasted_tgav_anomaly_all_pangeo_list_models.csv", 
+                      stringsAsFactors = FALSE) %>% dplyr::select(-X)
+
+
+# A chunked smoothed tgav anomaly archive of data. This was tgav_data but several 
+# steps were taken in python to transform it into the "chunked" data and save it 
+# so that we do not have to repeate this process so many times. 
+archive_data <- read.csv('inputs/archive_data.csv', stringsAsFactors = FALSE)
+
+
+# This is chunked smooth tgav anomaly for a single model/experiment/ensemble
+# member, saved for our convenience. If you decide that you want to work with 
+# a different different target data subset you can subset it from the
+# archive_data data frame. This file is SSP245 Realization 1.
+target_data <- read.csv("inputs/target_data.csv", stringsAsFactors = FALSE)
+
+# For plotting
+tgav_data %>%  
+  filter(model == unique(target_data$model) & experiment == unique(target_data$experiment) & 
+           ensemble == unique(target_data$ensemble)) -> 
+  original_data1
+
+
+# target data 2: ssp245 realization 4
+archive_data %>%
+  filter(experiment == 'ssp245' & ensemble == 'r4i1p1f1') ->
+  target_data2
+ 
+# for plotting
+tgav_data %>%  
+  filter(model == unique(target_data2$model) & experiment == unique(target_data2$experiment) & 
+           ensemble == unique(target_data2$ensemble)) -> 
+  original_data2
+
+```
+
+
+# Define the archive for testing
+
+The full archive, including the two trajectories we are emulating:
+
```

Comment:
agree that it's counter-intuitive especially for ssp245.  My thinking was more that by the time we want to generate realizations for all SSP585 targets, if we remove all of those potential targets from the archive, our generated realizations will probably be terrible and look more like an rcp60, especially towards the end of the century, because that's the next closest option when all 585 runs are removed from the archive. So this was me thinking more 'ok if we say these target _points_ individually are still in the archive for us to pull from, there a chance we just replicate the target realization as one of the many recipes we could stitch, but probably not'

Reply:
👌